server/on-demand-entry-handler.js

```diff
@@ -0,0 +1,221 @@
+import DynamicEntryPlugin from 'webpack/lib/DynamicEntryPlugin'
+import { EventEmitter } from 'events'
+import { createServer } from 'http'
+import { join } from 'path'
+import resolvePath from './resolve'
+import touch from 'touch'
+import WebSocket from 'ws'
+
+const ADDED = Symbol()
+const BUILDING = Symbol()
+const BUILT = Symbol()
+
+export default async function onDemandEntryHandler (devMiddleware, compiler, {
+  dir,
+  dev,
+  maxInactiveAge = 1000 * 25
+}) {
+  const server = await getServer()
```

Comment:
> Even then, we would only take up one socket ? 
With HTTP2 yes it is. But it's not there always.

> I believe webpack uses WS?
No it's not. It uses server sent events(HMR middleware) + 2 HTTP requests.